app/server/src/models/enrollment/enrollment.model.js

```diff
@@ -0,0 +1,41 @@
+const mongoose = require("mongoose");
+
+const enrollmentSchema = new mongoose.Schema({
+  user_id: {
+    type: String,
+  },
+  course: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "Course",
+  },
+  created_at: {
+    type: Date,
+  },
+  is_active: {
+    type: Boolean,
+  },
+  notes: {
+    type: [mongoose.Schema.Types.ObjectId],
+    ref: "Note",
+  },
+  progress: {
+    type: Map,
+    of: Boolean,
+  },
+});
+
+const Enrollment = mongoose.model("Enrollment", enrollmentSchema);
+
+const createEnrollment = async (user_id, course_id) => {
+  var enrollment = new Enrollment({ user_id, course_id });
+  enrollment.created_at = Date.now();
```

Comment:
instead of using Date.now(), let's use timestamps of mongodb, so that updatedAt, createdAt fields are automatically generated by the db itself

Reply:
changed exactly like you said!